lib/muffin_man/notifications/v1.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module MuffinMan
+  module Notifications
+    require "json"
+    require "sp_api_helpers"
+    class V1 < SpApiClient
+      NOTIFICATION_PATH = "/notifications/v1"
+      NOTIFICATION_SCOPE = "sellingpartnerapi::notifications"
+
+      def create_destination(arn, name, params = {})
+        @local_var_path = "#{NOTIFICATION_PATH}/destinations"
+        params = params.transform_keys(&:to_s)
+        @scope = NOTIFICATION_SCOPE
+        destination_params = { "resourceSpecification" => { "sqs" => { "arn" => arn } }, "name" => name }
+        unless params["region"].nil? || params["account_id"].nil?
+          destination_params["resourceSpecification"].merge!("eventBridge" => { "region" => params["region"],
+                                                                                "accountId" => params["account_id"] })
+        end
+        @request_body = destination_params
+        @request_type = "POST"
+        call_api
+      end
+
+      def get_destinations(params = {})
+        @local_var_path = "#{NOTIFICATION_PATH}/destinations"
+        @scope = NOTIFICATION_SCOPE
+        @query_params = sp_api_params(params)
+        @request_type = "GET"
+        call_api
+      end
+
+      def get_destination(destination_id)
+        @local_var_path = "#{NOTIFICATION_PATH}/destinations/#{destination_id}"
+        @scope = NOTIFICATION_SCOPE
+        @request_type = "GET"
+        call_api
+      end
+
+      def create_subscription(notification_type, params = {})
```

Comment:
notification_type and destination_id are required parameters, so I feel we should accept destination_id as a separate argument instead of accepting that from params. Thoughts?

Reply:
As per documentation, destination_id is an optional filed so kept it as in params.
Ref : https://developer-docs.amazon.com/sp-api/docs/notifications-api-v1-reference#createsubscriptionrequest